File: 28-implement-strstr/28-implement-strstr.java
class Solution {
    public static int strStr(String haystack, String needle) {
        
      int ans = -1;
      for (int i = 0; i < haystack.length(); i++) {
          int j = 0;
          while (haystack.charAt(i + j) == needle.charAt(j)) { // a == a
              if (j == needle.length() - 1) { // 0 != 2
                 return i;
              }
              if (i + j == haystack.length() - 1) {
                 return -1;
              }
              j++; //1
          }
      }     
      return ans;
    }
}

File: 28-implement-strstr/28-implement-strstr.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
    if (needle.length == 0) { return 0 }
    let index = -1
    
    for (let i = 0; i < haystack.length; i++) {
        if (haystack[i] == needle[0]) {
            for (let j = 0; j < needle.length; j++) {
                if (haystack[i + j] !== needle[j]) {
                    break
                }
                
                //base case
                if (j === needle.length - 1) {
                    return i
                }
            }    
        }
    }
    return index
};
console.log('expect 4, actual: ', strStr('mississippi', 'issip'))